docs/code/MlNetHighLevelConcepts.md

```diff
@@ -0,0 +1,151 @@
+# ML.NET high-level concepts
+
+In this document, we give a brief overview of the ML.NET high-level concepts. This document is mainly intended to describe the *model training* scenarios in ML.NET, since not all these concepts are relevant for the more simple scenario of *prediction with existing model*.
+
+## List of high-level concepts
+
+This document is going to cover the following ML.NET concepts:
+
+- *Data*, represented as an `IDataView` interface.
+  - In ML.NET, data is very similar to a SQL view: it's a lazily-evaluated, immutable, cursorable, heterogenous, schematized dataset. 
+  - An excellent document about the data interface is [IDataView Design Principles](IDataViewDesignPrinciples.md).
+- *Transformer*, represented as `ITransformer` interface.
+  - In one sentence, a transformer is a component that takes data, does some work on it, and return new 'transformed' data.
+  - For example, you can think of a machine learning model as a transformer that takes features and returns predictions.
```

Comment:
Just to make this even clearer, as model is a special case, it might be worth giving a data processing example such as "taking text data and outputting a numeric representation" or something like that

Reply:
I agree on that, let’s comment two samples, the typical pure data transformation such as dictionarizing or featurizing, then second example, more particular, the model..
I’d wish a model could be a specific different type instead of a transformer. Based on feedback from devs it is confusing for them to call a model=transformer.

Reply:
that's what I curremtly do.

---
In reply to: [221321970](https://github.com/dotnet/machinelearning/pull/960#discussion_r221321970) [](ancestors = 221321970)